### PR TITLE
Added unit test for wallet/storage.rs

### PR DESCRIPTION
**Pull request by Shourya742**
This PR is for issue: #20 

This is a micro PR for a series of PR for wallet module.

<details close>
<summary>All unit tests are passing; below, you will find the attached summary</summary>
<br>
Screenshot 2023-11-10 at 4 04 48 PM
</details>

### CODE REVIEW

File: src/wallet/storage.rs [hunk 1]
```diff
@@ -157,3 +157,106 @@ impl FileData {
...
+        )
+        .unwrap();
+
+        original_wallet_store.write_to_disk(&file_path).unwrap();
+
+        let read_wallet = WalletStore::read_from_disk(&file_path).unwrap();
+
+        assert_eq!(read_wallet.wallet_name, original_wallet_store.wallet_name);
+        assert_eq!(read_wallet.network, original_wallet_store.network);
+    }
```
Comment (rajarshimaitra) [inline src/wallet/storage.rs:214]: These two could be combined into one. read/write test already implies init works too. Keep it in a single test functions. and assert for both these cases. You would wanna assert for all the fields, that maximizes coverage ratio..

File: src/wallet/storage.rs [hunk 2]
```diff
@@ -157,3 +157,106 @@ impl FileData {
...
+
+    #[test]
+    fn test_try_from_file_data() {
```
Comment (rajarshimaitra) [inline src/wallet/storage.rs:217]: same comment as above for filedata tests..

File: src/wallet/storage.rs [hunk 3]
```diff
@@ -157,3 +157,106 @@ impl FileData {
...
+mod tests {
+    use super::*;
+    use bip39::Mnemonic;
+    use bitcoin::Network;
+    use bitcoind::tempfile::tempdir;
+    // Generate a dummy mnemonic for testing
+
+    fn generate_dummy_mnemonic() -> String {
+        Mnemonic::generate(12).unwrap().to_string()
+    }
```
Comment (rajarshimaitra) [inline src/wallet/storage.rs:171]: when you have a single line, no point in putting a function.

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
Good start.. Few suggestions..

File: src/wallet/storage.rs (hunk 3 above)
Comment (Shourya742) [reply src/wallet/storage.rs:171]: Initially I thought of adding a couple of things but then reduced it to Mnemonic generator only... Peace, I will change it.

File: src/wallet/storage.rs (hunk 2 above)
Comment (Shourya742) [reply src/wallet/storage.rs:217]: ok

File: src/wallet/storage.rs (hunk 1 above)
Comment (Shourya742) [reply src/wallet/storage.rs:214]: Yes, make sense... While writing test for init methods I was aware of it but wanted your confirmation. Cool I will make changes.

File: src/wallet/storage.rs [hunk 4]
```diff
@@ -157,3 +157,103 @@ impl FileData {
...
+        assert_eq!(read_wallet.wallet_name, original_wallet_store.wallet_name);
+        assert_eq!(read_wallet.network, original_wallet_store.network);
+        assert_eq!(
+            read_wallet.external_index,
+            original_wallet_store.external_index
+        );
+        assert_eq!(
+            read_wallet.incoming_swapcoins,
+            original_wallet_store.incoming_swapcoins
+        );
```
Comment (rajarshimaitra) [inline src/wallet/storage.rs:196]: Stupid me, I missed that you can derive `ParitalEq, Eq` for `WalletStore`. This whole assert series could be just 
`assert_eq!(read_wallet, original_wallet_store);`. 

Including all the asserts below.

File: src/wallet/storage.rs [hunk 5]
```diff
@@ -157,3 +157,103 @@ impl FileData {
...
+        assert_eq!(
+            read_wallet.offer_maxsize,
+            original_wallet_store.offer_maxsize
+        );
+        assert_eq!(read_wallet.master_key, original_wallet_store.master_key);
+    }
+
+    #[test]
+    fn test_file_data_save_and_load() {
```
Comment (rajarshimaitra) [inline src/wallet/storage.rs:217]: Actually, an `WalletStore` read/write implies `FileData` read/write. If I am not missing anything, then this is already covered so just one test for this module is enough.

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
Concept ACK.. 

Few changes needed..

File: src/wallet/storage.rs (hunk 4 above)
Comment (Shourya742) [reply src/wallet/storage.rs:196]: My bad... I forgot to ask...

**Review by rajarshimaitra — APPROVED** [review]
ACK 758e4de07f7a804a2a7258615e3678b8820bfd51

**Pull request merged**